src/librustc_codegen_llvm/debuginfo/mod.rs

```diff
@@ -483,7 +488,12 @@ impl DebugInfoMethods<'tcx> for CodegenCx<'ll, 'tcx> {
                     // so avoid methods on other types (e.g., `<*mut T>::null`).
                     match impl_self_ty.kind {
                         ty::Adt(def, ..) if !def.is_box() => {
-                            Some(type_metadata(cx, impl_self_ty, rustc_span::DUMMY_SP))
+                            // Again, only create type information if full debuginfo is enabled
```

Comment:
Please remove the "again". Nowhere in this file is there another occurence of this check.

Reply:
The comment is verbatim copy-pasted from a few lines up, which also has the verbatim condition `if cx.sess().opts.debuginfo == DebugInfo::Full`.